packages/@aws-cdk/aws-events/lib/rule.ts

```diff
@@ -138,6 +140,58 @@ export class Rule extends Resource implements IRule {
 
     const roleArn = targetProps.role ? targetProps.role.roleArn : undefined;
 
+    if (targetProps.target) {
+      const targetStack = Stack.of(targetProps.target);
+      const sourceStack = Stack.of(this);
+      const sourceAccount = sourceStack.account;
+      if (targetStack.account !== sourceAccount) {
+        // cross-account event - strap in, this works differently than regular events!
+        // based on:
+        // https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/CloudWatchEvents-CrossAccountEventDelivery.html
+
+        // the _actual_ target is just the event bus of the target's account
+        this.targets.push({
+          id,
+          arn: `arn:aws:events:${targetStack.region}:${targetStack.account}:event-bus/default`,
+        });
+
+        // Grant the source account permissions to the event bus of the target account.
+        // Do it in a separate stack, as it needs to be created before the rule containing the above event bus target
+        // (CloudWatch verifies whether you have permissions to the target on rule creation),
+        // but it's very likely the target stack has a dependency on the source (= this) stack
+        // (that's the case with CodePipeline, for example)
+        const stackId = `EventBusPolicy-${sourceAccount}`;
+        let eventBusPolicyStack: Stack = this.node.tryFindChild(stackId) as Stack;
+        if (!eventBusPolicyStack) {
+          eventBusPolicyStack = new Stack(this, stackId, {
+            env: {
+              account: targetStack.account,
+              region: targetStack.region,
+            },
+            stackName: `${targetStack.stackName}-EventBusPolicy-support-${sourceAccount}`,
+          });
+          new CfnEventBusPolicy(eventBusPolicyStack, `GivePermToOtherAccount`, {
+            action: 'events:PutEvents',
+            statementId: 'MySid',
+            principal: sourceAccount,
+          });
+          // deploy this before the source (= this) stack
+          sourceStack.addDependency(eventBusPolicyStack);
+        }
+
+        // The actual rule lives in the target stack.
+        // Other than the account, it's identical to this one
+        new Rule(targetStack, `${this.node.uniqueId}-${targetProps.id}`, {
```

Comment:
Can you add some  variables so that `this.target.push()` and `new Rule()` only occurs once  in this entire method? I think it might be more readable.

Reply:
Ah okay. Nevermind then.